Framework/Algorithms/src/SortXAxis2.cpp

```diff
@@ -0,0 +1,146 @@
+#include "MantidAlgorithms/SortXAxis2.h"
+
+using namespace Mantid::Kernel;
+using namespace Mantid::API;
+
+namespace Mantid {
+namespace Algorithms {
+
+DECLARE_ALGORITHM(SortXAxis)
+
+const std::string SortXAxis::name() const { return "SortXAxis"; }
+
+int SortXAxis::version() const { return 2; }
+
+const std::string SortXAxis::category() const {
+  return "Transforms\\Axes;Utility\\Sorting";
+}
+
+const std::string SortXAxis::summary() const {
+  return "Clones the input MatrixWorkspace(s) and orders the x-axis in an "
+         "ascending or descending fashion.";
+}
+
+void SortXAxis::init() {
+  declareProperty(make_unique<WorkspaceProperty<MatrixWorkspace>>(
+                      "InputWorkspace", "", Direction::Input),
+                  "Input Workspace");
+
+  declareProperty(make_unique<WorkspaceProperty<MatrixWorkspace>>(
+                      "OutputWorkspace", "", Direction::Output),
+                  "Sorted Output Workspace");
+
+  auto orderingValues = std::vector<std::string>({"Ascending", "Descending"});
+  auto orderingValidator =
+      boost::make_shared<StringListValidator>(orderingValues);
+  declareProperty("Ordering", orderingValues[0], orderingValidator,
+                  "Ascending or descending sorting", Direction::Input);
+}
+
+void SortXAxis::exec() {
+
+  MatrixWorkspace_const_sptr inputWorkspace = getProperty("InputWorkspace");
+  MatrixWorkspace_sptr outputWorkspace = inputWorkspace->clone();
+
+  for (auto specNum = 0u; specNum < inputWorkspace->getNumberHistograms();
+       specNum++) {
+
+    const auto sizeOfX = inputWorkspace->x(specNum).size();
```

Comment:
Some input validation is missing: contain the input workspaces point data as required? You can override a method `validateInputs`. And do all spectra have the same size? This is not required, all spectra can be treated independently.

Reply:
Ah, I see that all spectra are treated independently.

Reply:
Is this still an issue, or is it fine?

Reply:
I will commit the input validation & modified tests since I think it's needed. Sorting of histograms is possible with the Python algorithm and I think it's dangerous to do that! Now, this algorithm works as described by its documentation.